internal/controller/storageclient_controller.go

```diff
@@ -207,6 +207,7 @@ func (r *StorageClientReconciler) reconcilePhases() (ctrl.Result, error) {
 
 	if storageClientResponse.SystemAttributes != nil {
 		r.storageClient.Status.InMaintenanceMode = storageClientResponse.SystemAttributes.SystemInMaintenanceMode
+		r.storageClient.Status.MirrorEnabled = storageClientResponse.SystemAttributes.MirrorEnabled
```

Comment:
Because we are talking about the status section (where we reflect the actual state and not the desired state) I am not sure it is correct to reflect that the client is mirror-enabled before validating that the omap generator is running. In addition shouldn't we reflect similar things for other clients? shouldn't this be more granular?

Reply:
Here, I want to convey that the client is enabled for mirroring on the provider side. Hence storing the actual state in the Status, the desired state would be in the storage-client-mapping in the Provider side

Reply:
We have an issue here, status should not be treated as desired state and should be considered as something that can be deleted at any point in time.